### PR TITLE
Add README for tree-based models

### PR DESCRIPTION
**Pull request by qbc2016**
As the title says.

### CODE REVIEW

File: federatedscope/vertical_fl/README.md [hunk 1]
```diff
@@ -10,4 +11,170 @@ More details of the provided example can be found in [Tutorial](https://federate
...
+  # 0 indicates using the entire dataset (disable debug mode)
+
+# Evaluation related options
+eval:
+	# Frequency of evaluation
+  freq: 3
+  best_res_update_round_wise_key: test_loss
+```
+Users can specify ```model.type``` and ```vertical.algo``` to use different models. For examples,:
+-  XGBoost: `model.type = xgb_tree`,and `vertical.algo = 'xgb'`
```
Comment (qbc2016) [inline federatedscope/vertical_fl/README.md:76]: please add a space between the comma and 'and' (also for the following two lines)

File: federatedscope/vertical_fl/README.md [hunk 2]
```diff
@@ -10,4 +11,170 @@ More details of the provided example can be found in [Tutorial](https://federate
...
+
+- `adjusting` means we map the data into the integers between $[lb, ub]$, and then partition $[lb, ub]$ into $pb$ buckets evenly. For a value $x$ inside the $m$-th bucket, we first randomly select a bucket $i$ with probability $$p=\frac{e^{-|m-i|\cdot\epsilon_{p\ r\ t}/2}}{\sum_{j\in[lb, ub]}e^{-|m-j|\cdot\epsilon_{p\ r\ t}/2}}$$, then we randomly select a value $v$ in the selected bucket with probability $$p=\frac{e^{-|x-v|\cdot\epsilon_{n\ e\ r}/2}}{\sum_{j\in[lb, ub]} e^{-|x-j|\cdot\epsilon_{n\ e\ r}/2}}$$.  
+When $lb$ and $ub$ are close, $\epsilon, \epsilon_{prt}, \epsilon_{ner}$ and $pb$ are small, the strength for privacy protection is strong but the model utility can be affected.
+
+In `protect_args`, you can also add `bucket_num` to accelerate the training which is similar to the hist algorithm in XGBoost.
+
+The above two protection method are proposed in  "FederBoost: Private Federated Learning for
```
Comment (qbc2016) [inline federatedscope/vertical_fl/README.md:111]: ‘method’ -> 'methods'

File: federatedscope/vertical_fl/README.md [hunk 3]
```diff
@@ -10,4 +11,170 @@ More details of the provided example can be found in [Tutorial](https://federate
...
+
+`vertical.eval: 'ss'` (coming soon!) means for each tree, task party first SS the weight of each leaf node. And during inference, the indicator vectors of left and right children are also secret shared. By SS multiplication, at the end, each party will get a secret shared piece of the testing results. Then task party receives the pieces of data parties to reveal the exact result. The main advantage is that the indicator vectors are masked. 
+
+`vertical.eval: 'he'` means for each tree, each party locally get the leaf vector. Here, a leaf vector is a $0$-$1$ vector of length equals to the number of leaf node plus 1, where $0$ means that the sample must not in this leaf node and $1$ otherwise. Task party put the weight of the leaf node into the leaf vector where the corresponding coordinate has a $1$ and encrypts it by PHE and sends it to one data party. The data party  do the dot production between his own leaf vector with encrypted vector, and sends it to the next data party. The next data party dose exactly the same thing. Finally, the last data party sums up each component of the vector and sends it to task party. Task party decrypts it to get the testing result.
```
Comment (qbc2016) [inline federatedscope/vertical_fl/README.md:137]: '$0$-$1$' is presented as '0-$1$', which is weird.

File: federatedscope/vertical_fl/README.md [hunk 4]
```diff
@@ -10,4 +11,170 @@ More details of the provided example can be found in [Tutorial](https://federate
...
+  splits: [0.8, 0.2]  # splits for training and testing
+dataloader:
+  type: raw  # Personalized DataLoader
+  batch_size: 4177
+criterion:
+  type: RegressionMSELoss  # CrossEntropyLoss, for binary classification
+
+# Trainer related options
+trainer:
+	# Trainer type
```
Comment (qbc2016) [inline federatedscope/vertical_fl/README.md:55]: please move it to a suitable place

File: federatedscope/vertical_fl/README.md [hunk 5]
```diff
@@ -10,4 +11,170 @@ More details of the provided example can be found in [Tutorial](https://federate
...
+  # and the other has feature 4~7
+  feature_subsample_ratio: 1.0 # default = 1.0 
+  # the proportion of the numbers of features used for training per user.
+  algo: 'xgb' # 'xgb' or 'gbdt' or 'rf'
+  data_size_for_debug: 0  # use a subset for debug in vfl,
+  # 0 indicates using the entire dataset (disable debug mode)
+
+# Evaluation related options
+eval:
+	# Frequency of evaluation
```
Comment (qbc2016) [inline federatedscope/vertical_fl/README.md:71]: please move it to a suitable place
